File: week6/pset6/sentimental-readability/readability.py
import string

text = str(input("Text: "))
textl = text.lower()
ascii = string.ascii_lowercase
lcount = 0
wcount = 1
scount = 0

for i in range(len(text)):
    for letter in ascii:
        if letter == textl[i]:
            lcount += 1


for i in range(len(text)):
    if textl[i] == " ":
        wcount += 1

for i in range(len(text)):
    if textl[i] == "." or textl[i] == "!" or textl[i] == "?":
        scount += 1

L = lcount * 100 / wcount
S = scount * 100 / wcount
grade = round(0.0588 * L - 0.296 * S - 15.8)

if grade < 1:
    print("Before Grade 1")
    quit(0)

if grade >= 16:
    print("Grade 16+")
    quit(0)

print(f"Grade {grade}")

File: week2/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentence(string text);
int calc_grade(int words, float letters, float sentences);


int main(void)
{
    // Promt and print User input
    string input = get_string("Text: ");

    // calc letters
    int letters = count_letters(input);

    // calc words
    int words = count_words(input);

    // calc sententences
    int sentences = count_sentence(input);

    // print Grade
    int grade = calc_grade(words, letters, sentences);
    if (grade >= 16)
    {
        printf("Grade 16+");
    }

    if (grade < 1)
    {
        printf("Before Grade 1");
    }

    if (grade >= 1 && grade < 16)
    {
        printf("Grade %i", grade);
    }



    printf("\n");

}


int count_letters(string text)
{
    int calcl = 0;
    int len = strlen(text);
    for (int i = 0; i < len; i++)
    {
        int letter = toupper(text[i]);
        if (letter >= 65 && letter <= 90)
        {
            calcl = calcl + 1;
        }
    }
    return calcl;
}


int count_words(string text)
{
    int calcl = 0;
    int len = strlen(text);
    for (int i = 0; i < len; i++)
    {
        if (text[i] == 32)
        {
            calcl = calcl + 1;
        }
    }
    return calcl + 1;

}

int count_sentence(string text)
{
    int calcl = 0;
    int len = strlen(text);
    for (int i = 0; i < len; i++)
    {
        if ((text[i] == 46) || (text[i] == 63) || (text[i] == 33))
        {
            calcl = calcl + 1;
        }
    }
    return calcl;
}

int calc_grade(int words, float letters, float sentences)
{
    float L = letters * 100 / words;
    float S = sentences * 100 / words;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int rounded = round(index);
    return rounded;
}